### DISCUSSION TITLE
Mission name change

### DISCUSSION THREAD

**Discussion by ChuckStarchaser** [endless-sky/endless-sky#7533] [Ideas]
The simple idea: Allow mission scripts to change the mission name, to better reflect what the next step is.
For example, if a mission named "Recover Probe" is to recover a probe and bring it back, once the player recovers the probe, the game script could change the mission's name to "Bring Back Probe".
Reason:  To have the name still read "Recover Probe" can be confusing, if you already have it.  Even more confusing if you land on a planet to verify it is in your cargo, and find out it isn't.
Alternative solution:  Allow missions to have a Name and a Stage;  and display both in the Missions panel.  This way, the mission title could be less instructional sounding, for example,

` Arach Lady's Lost Probe
`
and have 4 stages named:

```
Search Planet A
Visit Pirate Base B
Search Pirate Quicksilver
Bring Back the Probe

```
And so the mission panel could read,

`Arach Lady's Lost Probe  --  Bring Back the Probe
`

**Comment by warp-core** [comment]
This is definitely something that could be added and seems like a reasonable addition to me.

Also, for future reference, specific feature requests like ought to be issues.

**Reply by Zireael07** [reply]
I know issues can be transferred to discussion - can they the other way round?

**Reply by Hecter94** [reply]
Yes they can.

**Comment by ChuckStarchaser** [comment]
Issue 7540.
